src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexLexer.cs

```diff
@@ -0,0 +1,448 @@
+                case '\u000C':
+                case '\u000D':
+                case ' ':
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        public RegexToken? TryScanEscapeCategory()
+        {
+            var start = Position;
+            while (Position < Text.Length &&
+                   IsEscapeCategoryChar(this.CurrentChar))
+            {
+                Position++;
+            }
+
+            if (Position == start)
+            {
+                return null;
+            }
+
+            var token = CreateToken(RegexKind.EscapeCategoryToken, ImmutableArray<RegexTrivia>.Empty, GetSubPattern(start, Position));
+            var category = token.VirtualChars.CreateString();
+
+            if (!RegexCharClass.IsEscapeCategory(category))
+            {
+                token = token.AddDiagnosticIfNone(new EmbeddedDiagnostic(
+                    string.Format(WorkspacesResources.Unknown_property_0, category),
+                    token.GetSpan()));
+            }
+
+            return token;
+        }
+
+        private static bool IsEscapeCategoryChar(VirtualChar ch)
+            => ch == '-' ||
+               (ch >= 'a' && ch <= 'z') ||
+               (ch >= 'A' && ch <= 'Z');
+
+        public RegexToken? TryScanNumber()
+        {
+            if (Position == Text.Length)
+            {
+                return null;
+            }
+
+            if (!IsDecimalDigit(this.CurrentChar))
+            {
+                return null;
+            }
+
+            const int MaxValueDiv10 = int.MaxValue / 10;
+            const int MaxValueMod10 = int.MaxValue % 10;
+
+            var value = 0;
+            var start = Position;
+            var error = false;
+            while (Position < Text.Length && this.CurrentChar is var ch && IsDecimalDigit(ch))
+            {
+                Position++;
+
+                unchecked
+                {
+                    var charVal = ch - '0';
+                    if (value > MaxValueDiv10 || (value == MaxValueDiv10 && charVal > MaxValueMod10))
+                    {
+                        error = true;
+                    }
+
+                    value *= 10;
+                    value += charVal;
+                }
+            }
+
+            var token = CreateToken(RegexKind.NumberToken, ImmutableArray<RegexTrivia>.Empty, GetSubPattern(start, Position));
+            token = token.With(value: value);
+
+            if (error)
+            {
+                token = token.AddDiagnosticIfNone(new EmbeddedDiagnostic(
+                    WorkspacesResources.Capture_group_numbers_must_be_less_than_or_equal_to_Int32_MaxValue,
+                    token.GetSpan()));
+            }
+
+            return token;
+        }
+
+        public RegexToken? TryScanCaptureName()
+        {
+            if (Position == Text.Length)
+            {
+                return null;
+            }
```

Comment:
Should we also bail out if the current char isn't a word char?

Reply:
#resolved